### PR TITLE
feat: Mobile app in footer

### LINKED ISSUE #51630: Advertise the freeCodeCamp mobile app for mobile devices.
### Is your feature request related to a problem? Please describe.

It would be nice to Advertise the freeCodeCamp mobile app for mobile users. This will improve their experience. Currently what I am thinking :

1. A pop up:
"It seems you are using a mobile device. Try the freeCodeCamp Moible App." This would be shown once on every block for the available superblocks.

2. Advertising the app in the footer like this:
image

3. A link in the Navbar to a seperate page that links the IOS and Android version. With a paragraph of what is features. 

### Describe the solution you'd like

-

### Describe alternatives you've considered

-

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by arjunpras**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #51630

For Bigger Screens:

large screen

mobile screen

### CODE REVIEW

File: client/src/components/Footer/footer.css [hunk 1]
```diff
@@ -111,6 +111,32 @@ p.footer-donation a:hover {
   margin: 0 0 3rem;
 }
 
+.mobile-apps {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  padding-top: 2rem;
+}
```
Comment (Sembauke) [inline client/src/components/Footer/footer.css:119]:
```suggestion
.mobile-apps {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 2rem;
}
```

File: client/src/components/Footer/footer.css [hunk 2]
```diff
@@ -111,6 +111,32 @@ p.footer-donation a:hover {
...
+
+.mobile-apps-container {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  align-items: center;
+  height: 100vh;
+  max-height: 7rem;
+}
```
Comment (Sembauke) [inline client/src/components/Footer/footer.css:128]:
```suggestion
.mobile-apps-container {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  width: 100%;
  max-width: 25rem;
}
```

File: client/src/components/Footer/footer.css [hunk 3]
```diff
@@ -126,6 +152,14 @@ p.footer-donation a:hover {
...
     font-size: 15;
   }
+
+  .mobile-apps-container {
+    flex-direction: row;
+    justify-content: space-around;
+    width: 100%;
+    max-width: 25rem;
+    height: auto;
+  }
```
Comment (Sembauke) [inline client/src/components/Footer/footer.css:162]:
```suggestion
  .mobile-apps-container {
    flex-direction: row;
  }
```

**Review by Sembauke — CHANGES_REQUESTED** [review]
Hey @arjunpras,

This is looking great already! Couple of notes and questions:

- We should change "Mobile Apps" to "Mobile App" as we only have one app we want to advertise. We should change this in the CSS-classes and the Header.

- Why are you nesting the image element in a anchor element?

**Comment by naomi-lgbt** [comment]
> Why are you nesting the image element in a anchor element?

It turns the image into a clickable link - click the image, get taken to the app store. :3

**Review by huyenltnguyen — COMMENTED** [review]
We should have test coverage for this feature.

The tests can be added to https://github.com/freeCodeCamp/freeCodeCamp/blob/main/e2e/footer.spec.ts.

File: client/src/components/Footer/index.tsx [hunk 4]
```diff
@@ -182,6 +184,19 @@ function Footer(): JSX.Element {
...
+            <h2 id='mobile-apps' className='col-header'>
+              {t('Mobile Apps')}
+            </h2>
+            <div className='mobile-apps-container'>
+              <a href='https://apps.apple.com/us/app/freecodecamp/id6446908151?itsct=apps_box_link&itscg=30200'>
+                <img src={appleMobileAppLogo} alt='apple store badge' />
+              </a>
+              <a href='https://play.google.com/store/apps/details?id=org.freecodecamp'>
+                <img src={androidMobileAppLogo} alt='google play badge' />
+              </a>
```
[the remark this answers is not in the document]
Comment (Sboonny) [reply client/src/components/Footer/index.tsx:197]: Shouldn't they be translatable?

File: client/src/components/Footer/index.tsx [hunk 5]
```diff
@@ -182,6 +184,19 @@ function Footer(): JSX.Element {
               </Link>
             </li>
           </ul>
+          <div className='mobile-apps'>
+            <h2 id='mobile-apps' className='col-header'>
+              {t('Mobile Apps')}
```
[the remark this answers is not in the document]
Comment (arjunpras) [reply client/src/components/Footer/index.tsx:189]: I have added mobile-app key in translations.json file only for English. I believe other languages do need this key too. so inorder to add this key, from where can I translate "mobile-app" to other language?

**Review by Sembauke — APPROVED** [review]

**Review by huyenltnguyen — COMMENTED** [review]
@arjunpras Thank you for the work!

There are a couple changes we should make still. I'm listing them here for transparency, but I'll handle the update myself tomorrow :)
- The links should open in a new tab and should probably be in a `ul`
- Playwright tests should be added
- The CSS could be reduced if we use layout components from `ui-components`. The `Spacer` component should also be used to ensure that we are using the standard spacing values

**Comment by arjunpras** [comment]
Thanks @huyenltnguyen for letting me know more optimised solution 👍

**Review by huyenltnguyen — APPROVED** [review]
@Sembauke I'm approving my own code here. My code changes doubled the size of the PR, but it's mainly because of the additional tests. I'll let you handle the merge and do yell if there is anything jumps out at you.

---
@arjunpras Thank you again for the PR and congratulations on your first contribution to this repo 🎉
We look forward to your next.

Happy contributing!

**Pull request merged**